source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1109,21 +1109,47 @@ updateRSFromPrimary
         return
 
     # Election ids are ObjectIds, see
-    # "using setVersion and electionId to detect stale primaries"
+    # "Using electionId and setVersion to detect stale primaries"
     # for comparison rules.
 
-    # Null values for both electionId and setVersion are always considered less than
-    if serverDescription.electionId > topologyDescription.maxElectionId or (
-        serverDescription.electionId == topologyDescription.maxElectionId
-        and serverDescription.setVersion >= topologyDescription.maxSetVersion
-    ):
-        topologyDescription.maxElectionId = serverDescription.electionId
-        topologyDescription.maxSetVersion = serverDescription.setVersion
+    if serverDescription.maxWireVersion >= 17:
+      # Null values for both electionId and setVersion are always considered less than
+      if serverDescription.electionId > topologyDescription.maxElectionId or (
+          serverDescription.electionId == topologyDescription.maxElectionId
+          and serverDescription.setVersion >= topologyDescription.maxSetVersion
+      ):
+          topologyDescription.maxElectionId = serverDescription.electionId
+          topologyDescription.maxSetVersion = serverDescription.setVersion
+      else:
+          # Stale primary.
+          # replace serverDescription with a default ServerDescription of type "Unknown"
+          checkIfHasPrimary()
+          return
     else:
-        # Stale primary.
-        # replace serverDescription with a default ServerDescription of type "Unknown"
-        checkIfHasPrimary()
-        return
+      # Maintain old comparison rules, namely setVersion is checked before electionId
+      # see "Using electionId and setVersion to detect stale primaries"
+      # for comparison rules.
+
+      if topologyDescription.maxSetVersion is not null \
+          and topologyDescription.maxElectionId is not null \
+          and (topologyDescription.maxSetVersion > description.setVersion
+              or topologyDescription.maxSetVersion == description.setVersion
+              and topologyDescription.maxElectionId
+              > description.electionId):
+
+          # Stale primary.
+          # replace description with a default ServerDescription of type "Unknown"
+
+          checkIfHasPrimary()
+          return
+
+      topologyDescription.maxElectionId = description.electionId
```

Comment:
this and below should fall under the legacy `else:` case.

Reply:
yea mixed indentation whoops, hopefully formatting fixed it